central/complianceoperator/v2/report/manager/complianceReportgenerator/report_gen_impl.go

```diff
@@ -167,19 +171,37 @@ func (rg *complianceReportGeneratorImpl) getDataforReport(req *ComplianceReportR
 			row.Profile = profiles[0].GetName()
 
 			// get remediation for the check result
-			q = search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorCheckName, c.GetCheckName()).AddExactMatches(search.ClusterID, c.GetClusterId()).ProtoQuery()
+			q = search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorCheckName, checkResult.GetCheckName()).AddExactMatches(search.ClusterID, checkResult.GetClusterId()).ProtoQuery()
 			remediations, err := rg.remediationDS.SearchRemediations(req.Ctx, q)
 			if err != nil {
 				return err
 			}
 			if len(remediations) > 0 {
```

Comment:
@SimonBaeumer confirmed there could be multiple remediations per check.  So this should be a list of remediations and not just a single one.